merkle/simple_tree.go

```diff
@@ -0,0 +1,76 @@
+/*
+Computes a deterministic minimal height merkle tree hash.
+If the number of items is not a power of two, some leaves
+will be at different levels. Tries to keep both sides of
+the tree the same size, but the left may be one greater.
+
+Use this for short deterministic trees, such as the validator list.
+For larger datasets, use IAVLTree.
+
+                        *
+                       / \
+                     /     \
+                   /         \
+                 /             \
+                *               *
+               / \             / \
+              /   \           /   \
+             /     \         /     \
+            *       *       *       h6
+           / \     / \     / \
+          h0  h1  h2  h3  h4  h5
+
+*/
```

Comment:
Can we add a warning here about pre-image attacks, or, more general a warning on how to use all this safely (in out context, but also in general). See for instance: https://github.com/bitcoin/bitcoin/blob/5961b23898ee7c0af2626c46d5d70e80136578d3/src/consensus/merkle.cpp#L9-L43

Reply:
I can take care of this later too (I would open an issue though referencing relevant comments)

Reply:
Would love it if you could pick this up! Thanks!

Reply:
https://github.com/tendermint/go-crypto/issues/107